packages/xod-client-electron/src/upload/reducer.js

```diff
@@ -1,15 +1,36 @@
 import * as R from 'ramda';
-import { SELECT_SERIAL_PORT } from '../upload/actionTypes';
+import client from 'xod-client';
+import { SELECT_SERIAL_PORT, UPLOAD } from '../upload/actionTypes';
 
 const initialState = {
   selectedSerialPort: null,
+  blocked: false,
+};
+
+const switchUploadBlock = (action, state) => {
+  const status = R.path(['meta', 'status'], action);
+
+  if (status === client.STATUS.STARTED) {
+    return R.assoc('blocked', true, state);
+  }
+  if (status === client.STATUS.SUCCEEDED || status === client.STATUS.FAILED) {
+    return R.assoc('blocked', false, state);
+  }
+
+  return state;
 };
 
 export default (state = initialState, action) => {
   switch (action.type) {
     case SELECT_SERIAL_PORT:
       return R.assoc('selectedSerialPort', action.payload.port, state);
 
+    case client.INSTALL_ARDUINO_DEPENDENCIES:
+      return switchUploadBlock(action, state);
+
+    case UPLOAD:
+      return switchUploadBlock(action, state);
```

Comment:
A potential race condition is here. Suggest tracking independently: instead of single `blocked` use `isInstallingArduinoDependencies` and `isUploading`.